eng/pipelines/publish-docker-image.yml

```diff
@@ -66,4 +81,40 @@ jobs:
                   containerRegistry: ${{ parameters.ContainerRegistry }}
                   repository: ${{ config.dockerRepo }}
                   command: push
-                  tags: ${{ stableTag }}
\ No newline at end of file
+                  tags: ${{ stableTag }}
+
+  - ${{ if and(parameters.ManifestDeployment, parameters.Publish) }}:
+    - ${{ each deployment in parameters.ManifestDeployment }}:
+      - job: container_build_${{ deployment.name }}
+        displayName: Build ${{ deployment.name }} Manifest
+        dependsOn:
+          - ${{ each config in parameters.DockerDeployments }}:
+            - container_build_${{ config.name }}
+        pool:
+          name: 'azsdk-pool-mms-ubuntu-2004-general'
+          vmImage: 'MMSUbuntu20.04'
+
+        steps:
+          - task: Docker@2
+            displayName: Login to ACR
+            inputs:
+              command: login
+              containerRegistry: ${{ parameters.ContainerRegistry }}
+
+          - pwsh: |
+              $configurations = '${{ convertToJson(parameters.DockerDeployments) }}' -replace '\\', '/'
+              $(Build.SourcesDirectory)/eng/pipelines/templates/scripts/get-docker-manifest-input.ps1 `
+                -DockerDeploymentJson $configurations `
+                -ContainerRegistry "${{ parameters.ContainerRegistry }}" `
+                -ImageTag "${{ parameters.ImageTag }}" `
+                -DevopsVariable "ManifestVariable" `
+            displayName: Generate Manifest Variable
+
+          - pwsh: |
+              docker manifest create ${{ parameters.ContainerRegistry }}.azurecr.io/${{ deployment.dockerRepo }}:${{ parameters.ImageTag }} $(ManifestVariable)
```

Comment:
Maybe this is just preference, but I think it would be better to return the manifest variable from the `get-docker-manifest-input.ps1` script and set it in the inline powershell right after that call, as opposed to passing in a variable name to set. I think that the more explicit and obvious devops variable setting can be (i.e. in the yaml) the less likely we have hidden issues later. It also ensures the powershell script and its authors don't need to have any knowledge of a devops context.